### PR TITLE
WPORG Plugins: use 1.2 endpoint and pass locale parameter.

### LINKED ISSUE #57635: WP.org Plugins don't have the correct language or all data
- Let's pass the locale in the wp.org plugins api request. https://api.wordpress.org/plugins/info/1.2/?action=query_plugins&request[locale]=es_ES

- Let's make sure that all data is fetched for the plugin details page (eg active installations number)

### PR DESCRIPTION
**Pull request by cpapazoglou**
#### Changes proposed in this Pull Request

* uses latest `https://api.wordpress.org/plugins/info/1.2/` instead of `https://api.wordpress.org/plugins/info/1.1/`. The new endpoint has more data like "active installations" and also accepts only GET requests
* pass the locale depending on current user locale in both plugins list and plugin details requests
* refactor away the specific `pluginRequest` and use `getRequest` 
* fetches only data needed

#### Testing instructions

* on a language different than English visit:
* plugins list `/plugins`, perform a search, plugin details
* make sure that the results are translated. The featured list on the top is coming from WPCOM and thus only available in English. Some plugins may not have translations in the language selected

|Before | After|
|-------|------|
|SS 2021-11-05 at 15 15 00|SS 2021-11-05 at 15 14 31|

Fixes #57635

### CODE REVIEW

File: client/state/plugins/wporg/test/actions.js [hunk 1]
```diff
@@ -73,7 +73,7 @@ describe( 'WPorg Data Actions', () => {
 
 	test( 'FetchPluginList action should dispatch two actions and make a request', async () => {
 		await store.dispatch( fetchPluginsList( 'new', 1, undefined ) );
-		expect( store.dispatch ).toHaveBeenCalledTimes( 2 );
+		expect( store.dispatch ).toHaveBeenCalledTimes( 3 );
```
Comment (WBerredo) [inline client/state/plugins/wporg/test/actions.js:76]: Why does it need to call 3 actions now?  
If this is the right action the test description also needs to be updated.

**Review by WBerredo — COMMENTED** [review]
The behavior and the code seem correct to me.

But I am interested in your thoughts around:

> fetches all data for plugin information instead of a limited set

Does that mean we will be retrieving information that will not be displayed or having any other kind of usage? This can consume bandwidth, which can be somewhat critical to mobile users.

File: client/state/plugins/wporg/test/actions.js (hunk 1 above)
Comment (cpapazoglou) [reply client/state/plugins/wporg/test/actions.js:76]: Yes it needs 3 times now. I have updated the test description.

**Comment by cpapazoglou** [comment]
> The behavior and the code seem correct to me.
> 
> But I am interested in your thoughts around:
> 
> > fetches all data for plugin information instead of a limited set
> 
> Does that mean we will be retrieving information that will not be displayed or having any other kind of usage? This can consume bandwidth, which can be somewhat critical to mobile users.

I enabled all data until we see what we actually need in the plugin details page, thanks for reminding me c12344b!
🎉 Limiting the fields saved us 3kb from the New plugins request and 16kb from the Popular plugins request 🎉

**Review by WBerredo — APPROVED** [review]
All good 🥳

Well done limiting even more the fields and saving bandwidth.

**Pull request merged**